flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/TaskIOMetricGroup.java

```diff
@@ -182,14 +186,20 @@ public TimerGauge getHardBackPressuredTimePerSecond() {
         return hardBackPressuredTimePerSecond;
     }
 
+    public TimerGauge getIoBackPressuredTimePerSecond() {
+        return ioBackPressuredTimePerSecond;
+    }
+
     public long getBackPressuredTimeMsPerSecond() {
         return getSoftBackPressuredTimePerSecond().getValue()
-                + getHardBackPressuredTimePerSecond().getValue();
+                + getHardBackPressuredTimePerSecond().getValue()
+                + getIoBackPressuredTimePerSecond().getValue();
```

Comment:
Should this be added up to the backpressured time? IMO it fits better to the busyness.

If given subtask is spending tons of time waiting for the changelog, that's problem of that particular subtask, not a downstream subtask's problem. And that's more or less the distinction between busyness/backpressured states/times. In other words, as it is proposed currently, large waiting time for changelog would mess up with autoscalers. Subtask that's blocked waiting on changelog should be scaled up, so this time should be reported as busy time imo.

Moreover, as it is right now, aren't you sometimes accounting changelog time as backpressured (if this is "soft waiting for changelog" via availability future) and sometimes as busy ("hard waiting for changelog" when actually blocked in the changelog code).

Reply:
Thanks for the detailed explanation and analysis, treating `changelogTime` as busy time is better to understand. I have moved the `changelogBusyTime`  into the busy time.